service/frontend/accessControlledAdminHandler.go

```diff
@@ -251,6 +306,11 @@ func (a *AccessControlledWorkflowAdminHandler) PurgeDLQMessages(ctx context.Cont
 		APIName:    "PurgeDLQMessages",
 		Permission: authorization.PermissionAdmin,
 	}
+
+	if isManual(ctx) {
```

Comment:
it seems that it's worth moving `isManual` check into `isAuthorized`

Reply:
Since we change to store a pointer instead of a struct in Attributes, as we discussed, we don't need to check if it is manual any more. Because a pointer won't occupy too much space.